src/frequenz/sdk/_internal/_asyncio.py

```diff
@@ -28,6 +32,25 @@ async def cancel_and_await(task: asyncio.Task[Any]) -> None:
         pass
 
 
+async def run_forever(
+    async_callable: Callable[[], Coroutine[Any, Any, None]],
+    interval: timedelta = timedelta(seconds=1),
```

Comment:
In the future it might be nice to be able to pass a retry strategy. A good reason to move the `retry` module from `client-base` to `core`.

Reply:
ExponentialBackoff is hard to do with generic functions, because it needs to know if a previous attempt succeeded, so that it can reset its backoff interval.

This is easy to do with streams, we just reset it after every incoming message.  But for functions, we'd need something like a timer to reset the backoff interval, if the function hasn't failed for a certain interval, then we say it succeeded, for example.